### ISSUE TITLE
Thoughts

### ISSUE THREAD

**Issue by pine3ree** [slimphp/Slim#1198]
@codeguy 
Hi Josh,
i've spent some time thinking about where we are moving towards with this new psr-7 / middleware interoperability revolution.

I believe we are putting too much emphasis on request/response paradigm. That is the basis of a web application, but shouldn't be the goal.

As an example, slim-3-dev now have url handlers behaving like middlewares and returning responses.
I'm doing that too....but...

back in the (not so) old times....

we returned a string, (maybe calling "The View" render method with a template-vars parameters.
That string could be then, based on how we setup the view and on what rendering engin we(built and)  used, be decorated with a layout.
Now we can (and we also have full responsibility) of manipulating the whole response. That's cool, but that should be only an option not a must do in my opinion.
If we need to render a partial view into a layout, we would need to do a lot of extra work in every controller action (route handler) and then fill the response body with the result of rendering + decoration.
I believe this pattern is going distract us developers from the real goal of the action, the business logic.

In short, the old way offered a bit more automation (most of the time implemented by us as well) in building an application and let us focus on the real  job.

what's your opinion about this?

best regars

**Comment by akrabat** [comment]
Fixed in #1206.
